packages/store/addon/-private/system/fetch-manager.ts

```diff
@@ -0,0 +1,554 @@
+          error ||
+          new Error(
+            `Expected: '<${identifier.type}:${identifier.id}>' to be present in the adapter provided payload, but it was not found.`
+          )
+        );
+      }
+    }
+  }
+
+  _findMany(adapter: any, store: any, modelName: string, snapshots: Snapshot[], identifiers: RecordIdentifier[], optionsMap) {
+    let modelClass = store.modelFor(modelName); // `adapter.findMany` gets the modelClass still
+    let ids = snapshots.map((s) => s.id);
+    let promise = adapter.findMany(store, modelClass, ids, Ember.A(snapshots));
+    let label = `DS: Handle Adapter#findMany of '${modelName}'`;
+
+    if (promise === undefined) {
+      throw new Error('adapter.findMany returned undefined, this was very likely a mistake');
+    }
+
+    promise = guardDestroyedStore(promise, store, label);
+
+    return promise.then(
+      adapterPayload => {
+        assert(
+          `You made a 'findMany' request for '${modelName}' records with ids '[${ids}]', but the adapter's response did not have any data`,
+          !!payloadIsNotBlank(adapterPayload)
+        );
+        let serializer = serializerForAdapter(store, adapter, modelName);
+        let payload = normalizeResponseHelper(
+          serializer,
+          store,
+          modelClass,
+          adapterPayload,
+          null,
+          'findMany'
+        );
+        return payload;
+      },
+      null,
+      `DS: Extract payload of ${modelName}`
+    );
+  }
+
+
+  _processCoalescedGroup(seeking: { [id: string]: PendingFetchItem },  group: Snapshot[], adapter: any, optionsMap, modelName: string) {
+    let totalInGroup = group.length;
+    let ids = new Array(totalInGroup);
+    let groupedIdentifiers = new Array(totalInGroup);
+
+    for (var j = 0; j < totalInGroup; j++) {
+      groupedIdentifiers[j] = group[j]._identifier;
+      ids[j] = groupedIdentifiers[j].id;
+    }
+
+    let store = this._store;
+    if (totalInGroup > 1) {
+      this._findMany(adapter, store, modelName, group, groupedIdentifiers, optionsMap)
+        .then((payloads) => {
+          this.handleFoundRecords(seeking, payloads, groupedIdentifiers);
+        })
+        .catch((error) => {
+          this.rejectInternalModels(seeking, groupedIdentifiers, error);
+        });
+    } else if (ids.length === 1) {
+      var pair = seeking[groupedIdentifiers[0].id];
+      this._fetchRecord(pair);
+
+    } else {
+      assert(
+        "You cannot return an empty array from adapter's method groupRecordsForFindMany",
+        false
+      );
+    }
+  }
+
+  _flushPendingFetchForType(pendingFetchItems: PendingFetchItem[], modelName: string) {
+    let store = this;
+    let adapter = this._store.adapterFor(modelName);
+    let shouldCoalesce = !!adapter.findMany && adapter.coalesceFindRequests;
+    let totalItems = pendingFetchItems.length;
+    let identifiers = new Array(totalItems);
+    let seeking: { [id: string]: PendingFetchItem } = Object.create(null);
```

Comment:
```suggestion
    let seeking: { [id: string]: PendingFetchItem | undefined } = Object.create(null);
```

Reply:
I checked and the values will always be there